pkg/utils/validator/clusterresourceplacement.go

```diff
@@ -213,6 +225,40 @@ func validateClusterAffinity(clusterAffinity *placementv1beta1.ClusterAffinity,
 	return apiErrors.NewAggregate(allErr)
 }
 
+func validateTolerations(tolerations []placementv1beta1.Toleration) error {
+	allErr := make([]error, 0)
+	tolerationMap := make(map[placementv1beta1.Toleration]bool)
+	for _, toleration := range tolerations {
+		if toleration.Key != "" {
+			for _, msg := range validation.IsQualifiedName(toleration.Key) {
+				allErr = append(allErr, fmt.Errorf(invalidTolerationKeyErrFmt, toleration, msg))
+			}
+		}
+		switch toleration.Operator {
+		case corev1.TolerationOpExists:
+			if toleration.Value != "" {
+				allErr = append(allErr, fmt.Errorf(invalidTolerationErrFmt, toleration, "toleration value needs to be empty, when operator is Exists"))
+			}
+		case corev1.TolerationOpEqual:
+			if toleration.Key == "" {
+				allErr = append(allErr, fmt.Errorf(invalidTolerationErrFmt, toleration, "toleration key cannot be empty, when operator is Equal"))
+			}
+			if toleration.Value == "" {
+				allErr = append(allErr, fmt.Errorf(invalidTolerationErrFmt, toleration, "toleration value cannot be empty, when operator is Equal"))
+			} else {
+				for _, msg := range validation.IsValidLabelValue(toleration.Value) {
+					allErr = append(allErr, fmt.Errorf(invalidTolerationValueErrFmt, toleration, msg))
+				}
+			}
+		}
+		if _, exists := tolerationMap[toleration]; exists {
```

Comment:
nit, the idiomatic way to check a bool map is like this, 
```suggestion
		if tolerationMap[toleration]{
```